pkg/subroutines/creator_test.go

```diff
@@ -0,0 +1,354 @@
+				},
+				Spec: v1alpha1.AccountSpec{
+					Creator: &creator,
+				},
+				Status: v1alpha1.AccountStatus{
+					Namespace: &namespace,
+				},
+			},
+			in: &openfgav1.WriteRequest{
+				StoreId: "2",
+				Writes: &openfgav1.WriteRequestWrites{
+					TupleKeys: []*openfgav1.TupleKey{
+						{
+							Object:   "account:test-account",
+							Relation: "owner",
+							User:     "user:test-creator",
+						},
+					},
+				},
+			},
+			setupMocks: func(ctx context.Context, in *openfgav1.WriteRequest, out *openfgav1.WriteResponse, openFGAServiceClientMock *mocks.OpenFGAServiceClient, k8ServiceMock *mocks.K8Service) {
+				k8ServiceMock.EXPECT().GetFirstLevelAccountForNamespace(ctx, "openmfp-system").Return(&v1alpha1.Account{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "tenant2",
+						Namespace: "openmfp-system",
+					},
+					Spec: v1alpha1.AccountSpec{
+						Creator: &creator,
+					},
+					Status: v1alpha1.AccountStatus{
+						Namespace: &namespace,
+					},
+				}, nil)
+				openFGAServiceClientMock.EXPECT().ListStores(ctx, mock.Anything).Return(&openfgav1.ListStoresResponse{
+					Stores: []*openfgav1.Store{{Id: "2", Name: "tenant-tenant2"}},
+				}, nil)
+				openFGAServiceClientMock.EXPECT().Write(ctx, in).Return(out, assert.AnError).Once()
+			},
+		},
+		{
+			name: "should_not_write_owner_if_creator_is_nil",
+			account: v1alpha1.Account{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "tenant3",
+					Namespace: "openmfp-system",
+				},
+				Status: v1alpha1.AccountStatus{
+					Namespace: &namespace,
+				},
+			},
+			setupMocks: func(ctx context.Context, in *openfgav1.WriteRequest, out *openfgav1.WriteResponse, openFGAServiceClientMock *mocks.OpenFGAServiceClient, k8ServiceMock *mocks.K8Service) {
+			},
+		},
+		{
+			name: "should_not_write_owner_if_condition_exists",
+			account: v1alpha1.Account{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-account",
+					Namespace: "openmfp-system",
+				},
+				Status: v1alpha1.AccountStatus{
+					Namespace: &namespace,
+					Conditions: []metav1.Condition{
+						{
+							Type:    "CreatorSubroutine_Ready",
+							Status:  "True",
+							Reason:  "AccountReady",
+							Message: "Message is ready",
+						},
+					},
+				},
+			},
+			setupMocks: func(ctx context.Context, in *openfgav1.WriteRequest, out *openfgav1.WriteResponse, openFGAServiceClientMock *mocks.OpenFGAServiceClient, k8ServiceMock *mocks.K8Service) {
+			},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+
+			// setup mocks
+			openFGAServiceClientMock := mocks.NewOpenFGAServiceClient(t)
+			k8ServiceMock := mocks.NewK8Service(t)
+			if test.setupMocks != nil {
+				out := test.out
+				if out == nil {
+					out = &openfgav1.WriteResponse{}
+				}
+				test.setupMocks(test.ctx, test.in, out, openFGAServiceClientMock, k8ServiceMock)
+			}
+
+			routine := subroutines.NewCreatorSubroutine(openFGAServiceClientMock, k8ServiceMock, "openmfp-system")
```

Comment:
🟢 check for the operator result, and also if sentry and retry is set. 
We faced severe issues already, if retry is false